### PR TITLE
AuthMeBungee integration

### PR DESCRIPTION
**Pull request by sgdc3**
An injector related test is failing @ljacqu

### CODE REVIEW

File: src/main/java/fr/xephi/authme/service/BungeeService.java [hunk 1]
```diff
@@ -47,6 +65,34 @@ public void connectPlayer(Player player) {
...
+    /**
+     * Send the player's auth status to AuthMeBungee
+     *
+     * @param player The player
+     */
+    public void sendPlayerAuthStatus(Player player) {
+        final ByteArrayOutputStream bout = new ByteArrayOutputStream();
+        DataOutputStream out = new DataOutputStream(bout);
+
+        try {
```
Comment (sgdc3) [inline src/main/java/fr/xephi/authme/service/BungeeService.java]: you're right, i forgot to check if some parts of the code from authmebridge could be optimized ;)

**Comment by ljacqu** [comment]
Thanks @sgdc3, first thing I'll have a look at this evening :) On mobile, I can't even see the failing tests so I'm blind. I hope it's not a cyclic dependency.

File: src/main/java/fr/xephi/authme/service/BungeeService.java [hunk 2]
```diff
@@ -47,6 +65,34 @@ public void connectPlayer(Player player) {
...
+            }
+            out.writeUTF(player.getName());
+            new BukkitRunnable() {
+                @Override
+                public void run() {
+                    player.sendPluginMessage(plugin, "BungeeCord", bout.toByteArray());
+                }
+            }.runTaskAsynchronously(plugin);
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (ljacqu) [inline src/main/java/fr/xephi/authme/service/BungeeService.java]: Please log this exception as we do normally

Comment (sgdc3) [inline src/main/java/fr/xephi/authme/service/BungeeService.java]: Yep, forgot to change it to ConsoleLogger

**Comment by sgdc3** [comment]
@ljacqu Probably a cyclic dependency but the error message was related to the command handler :/

**Comment by ljacqu** [comment]
@sgdc3 Yeah, we have to create the ExecutableCommand implementations inside of the CommandHandler constructor, so a lot of classes (not only the commands but their dependencies) might be initialized at that moment. Unfortunately this results in ugly exceptions. Probably the cause, or the cause of the cause, is a cyclic dep. error

**Comment by ljacqu** [comment]
> Caused by: ch.jalu.injector.exceptions.InjectorException: Found cyclic dependency - already traversed 'class fr.xephi.authme.process.Management' (full traversal list: [class fr.xephi.authme.process.logout.AsynchronousLogout, class fr.xephi.authme.process.Management, class fr.xephi.authme.process.SyncProcessManager, class fr.xephi.authme.process.register.ProcessSyncPasswordRegister, class fr.xephi.authme.service.BungeeService])
	at ch.jalu.injector.InjectorImpl.validateInjectionHasNoCircularDependencies(InjectorImpl.java:291)

In other words, when the injector tries to instantiate the `Management` class, it needs to figure out its dependencies 
-> AsynchronousLogout is required 
-> which, in turn, requires SyncProcessManager 
-> which, in turn, requires ProcessSyncPasswordRegister 
-> which, in turn, requires BungeeService, 
-> and BungeeService now has Management as dependency 

= full circle.

graph

We need to find a way how to break this circle.

File: src/main/java/fr/xephi/authme/service/BungeeService.java (hunk 2 above)
Comment (DNx5) [inline src/main/java/fr/xephi/authme/service/BungeeService.java]: could we use guava [ByteArrayDataOutput](http://google.github.io/guava/releases/snapshot/api/docs/com/google/common/io/ByteArrayDataOutput.html) here? the difference is, it doesnt throw IOException

File: src/main/java/fr/xephi/authme/service/BungeeService.java [hunk 3]
```diff
@@ -47,6 +65,34 @@ public void connectPlayer(Player player) {
...
+     * @param player The player
+     */
+    public void sendPlayerAuthStatus(Player player) {
+        final ByteArrayOutputStream bout = new ByteArrayOutputStream();
+        DataOutputStream out = new DataOutputStream(bout);
+
+        try {
+            out.writeUTF("AuthMeBungee");
+            if(playerCache.isAuthenticated(player.getName())) {
+                out.writeUTF("LOGIN:");
```
Comment (DNx5) [inline src/main/java/fr/xephi/authme/service/BungeeService.java]: What if player hasn't registered on another server?

Comment (sgdc3) [inline src/main/java/fr/xephi/authme/service/BungeeService.java]: It must be the same database

**Comment by sgdc3** [comment]
I'm closing that as we should implement a more complete messaging protocol in order to sync other bukkit instances.

**Pull request closed without merging**